### DISCUSSION TITLE
Quickly find org file

### DISCUSSION THREAD

**Discussion by Maltimore** [nvim-orgmode/orgmode#810] [Q&A] [answered]
Hi,
I'm wondering how everyone here is navigating to their org files? My org files aren't all in one directory, so navigating to a specific org file is tedious. Ideally I'd like a way to fuzzy search all org files, i.e. all agenda files.

**Comment by breezykermo** [comment]
I use https://github.com/nvim-telescope/telescope.nvim, which is very configurable. You could configure it with the same Org agenda files list, I suppose, and map it to a hot key to search.

**Reply by Maltimore** [reply]
Thanks a lot!

**Comment by seflue** [comment, marked as answer]
I actually recommend the plugin [Telescope-orgmode.nvim](https://github.com/nvim-orgmode/telescope-orgmode.nvim). You can map a keybinding directly to find all your agenda files, but also all your headers in all your agenda files. You can also switch between these two modes.

I'm now maintaining this plugin and I also added features for convenient refiling and link insertion. The latter plays also nicely with [org-roam](https://github.com/chipsenkbeil/org-roam.nvim), when you use [id-based links](https://github.com/nvim-orgmode/orgmode/blob/master/DOCS.md#org_id_link_to_org_use_id). It uses [org_store_link](https://github.com/nvim-orgmode/orgmode/blob/master/DOCS.md#org_store_link) and [org_insert_link](https://github.com/nvim-orgmode/orgmode/blob/master/DOCS.md#org_insert_link) under the hood.

**Reply by Maltimore** [reply]
Great, this is perfect!

**Comment by kristijanhusak** [comment]
Not really a solution, but in case you want to build your own, you can get the list of all agenda file names like this:

```lua
local filenames = vim.tbl_map(function(file) return file.filename end, require('orgmode.api').load())
```

and then just pipe those through some fuzzy finder. https://github.com/nvim-orgmode/telescope-orgmode.nvim is probably the best option if you are using telescope.
